testing/unit_tests/test_testing.py

```diff
@@ -200,18 +200,18 @@ def test_inference_2d_microscopy(download_data_testing_test_files, transforms_di
         model.cuda()
     model.eval()
 
-    if not os.path.isdir(__output_dir__):
-        os.makedirs(__output_dir__)
+    if not __output_dir__.is_dir():
+        __output_dir__.mkdir(parents=True)
 
     preds_npy, gt_npy = imed_testing.run_inference(test_loader=test_loader,
                                                    model=model,
                                                    model_params=model_params,
                                                    testing_params=testing_params,
-                                                   ofolder=__output_dir__,
+                                                   ofolder=str(__output_dir__),
                                                    cuda_available=cuda_available)
 
-    assert len([x for x in os.listdir(__output_dir__) if x.endswith(".nii.gz")]) == len(test_lst)
-    assert len([x for x in os.listdir(__output_dir__) if x.endswith(".png")]) == 2*len(test_lst)
+    assert len([x for x in __output_dir__.iterdir() if x.name.endswith(".nii.gz")]) == len(test_lst)
+    assert len([x for x in __output_dir__.iterdir() if x.name.endswith(".png")]) == 2*len(test_lst)
```

Comment:
I think these can just use `Path().suffix` ?

Reply:
>>> from pathlib import Path
>>> a=Path("a.nii.gz")
>>> a.suffix
'.gz'

Reply:
I think we can potentially use [suffixes](https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.suffixes)

Reply:
I think checking a string with ends with is easier than checking a list